pyvista/core/filters/poly_data.py

```diff
@@ -2091,9 +2091,18 @@ def flip_normals(poly_data):
         if not poly_data.is_all_triangles():
             raise NotAllTrianglesError('Can only flip normals on an all triangle mesh.')
 
-        f = poly_data.faces.reshape((-1, 4))
-        f[:, 1:] = f[:, 1:][:, ::-1]
-        poly_data.faces = f
+        if _vtk.VTK9:
+            # use new connectivity API
+            faces = poly_data._connectivity_array.reshape(-1, 3)
+
+            # shuffle these in-place
+            # See: https://stackoverflow.com/a/33362288/3369879
+            faces[:, 0], faces[:, 2] = faces[:, 2], faces[:, 0].copy()
+
+        else:  # pragma: no cover
+            f = poly_data.faces.reshape((-1, 4))
+            f[:, 1:] = f[:, 1:][:, ::-1]
```

Comment:
And since we're optimizing like crazy, this line should at least match the newer version above. And I have another suggestion:
```py
    f = poly_data.faces
    f[1::4], f[3::4] = f[3::4], f[1::4].copy()
```
On numpy 1.21.1 and my laptop this is even faster for arrays sized 10_000 or smaller. Can you time this too on your system? (Since you couldn't reproduce my earlier timings).

Reply:
This approach is faster for all cases in 1.19.5 as well

```py
>>> import pyvista as pv
>>> import numpy as np
>>> mesh = pv.Sphere(phi_resolution=1000, theta_resolution=1000)

# 1,996,000 faces

>>> f = mesh.faces
>>> timeit f[1::4], f[3::4] = f[3::4], f[1::4].copy()
10.9 ms ± 73.1 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

>>> f = mesh.faces.reshape(-1, 4)
>>> timeit f[:, 1:] = f[:, 1:][:, ::-1]
15.4 ms ± 17.2 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)


# 160 faces
1.16 µs ± 5.12 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
2 µs ± 9.38 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```